Core/include/Acts/Utilities/detail/Line3DWithPartials.hpp

```diff
@@ -0,0 +1,64 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Utilities/AlgebraHelpers.hpp"
+#include "Acts/Utilities/ArrayHelpers.hpp"
+
+namespace Acts::detail {
+/// @brief Helper that to describe a line in 3D space. The line is described
+///        by a point on the line and a corresponding direction vector with unit
+///        length Additionally, the `Line3DWithPartials` holds the first and
+///        second derivative of the line with respect to the parameters `x0`,
+///        `y0`, `theta`, and `phi` (the parameters are defined in the enum
+///        `ParIndices`).
+template <std::floating_point T>
+class Line3DWithPartials {
```

Comment:
```suggestion
class Line3DWithPartialDerivatives {
```
at least to me "partials" did not really mean anything without the doc

same for the file name